Makefile

```diff
@@ -1,9 +1,53 @@
+SHELL = /bin/bash
 # We assume an active virtualenv for development
-install:
-	-@python3 setup.py -q develop --upgrade
+include requirements.txt
+PYENV_LOCATION = /usr/local/bin/pyenv
+PYENV_REGEX = .pyenv/shims
+VENV_NAME ?= .venv
+VENV_ACTIVATE_FILE = $(VENV_NAME)/bin/activate
+VENV_ACTIVATE = . $(VENV_ACTIVATE_FILE)
+VEPYTHON = $(VENV_NAME)/bin/python3i
+PYENV_ERROR = "\033[0;31mIMPORTANT\033[0m: Please install pyenv. \033[0;31m\n"
+PYENV_PATH_ERROR = "\033[0;31mIMPORTANT\033[0m: Please add $(HOME)/$(PYENV_REGEX) to your PATH env. \033[0;31m\n"
+PYENV_PREREQ_HELP = "\033[0;31mIMPORTANT\033[0m: please add \033[0;31meval \"\$$(pyenv init -)\"\033[0m to your bash profile and restart your terminal before proceeding any further.\n"
+VE_MISSING_HELP = "\033[0;31mIMPORTANT\033[0m: Couldn't find $(PWD)/$(VENV_NAME); have you executed make venv-create?\033[0m\n"
+
+prereq: requirements.txt
+	pyenv install $(PY34)
+	pyenv install $(PY35)
+	pyenv install $(PY36)
+	pyenv install $(PY37)
+	pyenv global system $(PY34) $(PY35) $(PY36) $(PY37)
+	-@ printf $(PYENV_PREREQ_HELP)
+
+venv-create:
+	@if [[ ! -f  $(PYENV_LOCATION) ]]; then \
```

Comment:
This failed on a CentOS 7 env I tried it in due to pyenv not being under `/usr/local...`.
Let's make this simpler using: `@if [[ ! -x $$(command -v pyenv) ]]; then \`

Reply:
Thank you for checking, I used your suggested check.